### DISCUSSION TITLE
Microsoft OpenJDK Java and Floating Point

### DISCUSSION THREAD

**Discussion by PowerUser1234** [microsoft/openjdk#171] [Q&A] [answered]
I would like to know how Microsoft OpenJDK Java performs.

-Is it compatible with Free Open Source Java libraries,
or other Java libraries, out there?

-Has Microsoft OpenJDK removed the phenomenon of Java
Floating Point denormal and pronormal values?
Consider the following Java code snippet:

**//----------------------------------------------------------
package JavaApplication;
import static java.lang.System.*;
public class Start
{
    public static void main(String ...args)
    {
    out.println("Program has started...");   
    double a = 0.1;   
    double b = 0.1;   
    double c = a*b;   
    out.println();   
    out.println(c);   
    out.println();   
    out.println(c == 0.01);   
    float d = 0.01F; 
    float e = 0.01F;
    float f = d*e;
    out.println();
    out.println(f);
    out.println();
    out.println(f == 0.01);   
    out.println(); 
    out.println("Program has Finished.");
    }}
//----------------------------------------------------------**

-What I need is a version of Java 17 that has the same
free-for-all-use license model as OpenJDK17, but has
removed the phenomenon of denormal and pronormal
values, of all defaulting arithmetic, systematically, 
involving float and double, and their objects.

_-Will Microsoft OpenJDK run the included code fragment,
outputting the generally expected results, 
printing true and not false?_

**Comment by karianna** [comment, marked as answer]
Hi @PowerUser1234 - I caught your message to the Adoptium working group and the same answer applies here really.  MSFT Build of OpenJDK is a downstream distribution of OpenJDK under the GPLv2+CE license.  We have no plans to change the default behavior of FP numbers at this time.

I'd recommend using hte Adoptium scripts to build your own custom OpenJDK binary in order to make that change for your own purposes.
